full-moon/tests/roblox_cases/pass/if_expression/source.lua

```diff
@@ -0,0 +1,6 @@
+local x = if foo then foo.x else 5
+local y = (if x then x.indices else create()):update(if shouldUpdate then information else defaults)
+local z = (if bar then foo.y else 5) :: number
+
+local a = if foo then foo.x elseif bar then bar.x else 5
```

Comment:
a nuance to capture:
```suggestion
local a = if foo then (foo.x :: number) elseif bar then bar.x()() else 5
```

Reply:
We should add this as another test case, instead of updating the existing one, but yeah!